scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -461,11 +456,15 @@
 
                                       :cook-expected-state/starting
                                       (case pod-synthesized-state-modified
-                                        :missing (launch-pod compute-cluster
-                                                             cook-expected-state-dict pod-name)
+                                        :missing (if (:pod-deleted? synthesized-state)
+                                                   (do
+                                                     (log/info "In compute cluster" name ", pod" pod-name
+                                                               "was deleted while it was expected starting")
+                                                     (handle-pod-killed compute-cluster pod-name))
```

Comment:
### Why?

There's a legitimate scenario where we launch a pod and then it gets deleted before it started running. In this case, we need to go down the `handle-pod-killed` path.